src/java.base/share/native/libjli/java.c

```diff
@@ -620,14 +614,36 @@ JavaMain(void* _args)
      * The main method is invoked here so that extraneous java stacks are not in
      * the application stack trace.
      */
-    if (!invokeStaticMainWithArgs(env, mainClass, mainArgs) &&
-        !invokeInstanceMainWithArgs(env, mainClass, mainArgs) &&
-        !invokeStaticMainWithoutArgs(env, mainClass) &&
-        !invokeInstanceMainWithoutArgs(env, mainClass)) {
-        ret = 1;
-        LEAVE();
+
+
+    helperClass = GetLauncherHelperClass(env);
+    isStaticMainField =
+        (*env)->GetStaticFieldID(env, helperClass, "isStaticMain", "Z");
```

Comment:
Nit: this line isn't long.  Can do in 1 line.

Same for line 623-624, 626-627.